### DISCUSSION TITLE
Static ID for unique Urls For Viewing Pdf in Browser

### DISCUSSION THREAD

**Discussion by MarcoScaletta** [James-Yu/LaTeX-Workshop#4337] [Q&A] [answered]
Hi!
I view pdfs in an external viewer, namely safari that opens the a tab with the following url
`http://<LOCAL_IP>:<PORT>/viewer.html?file=pdf..<ID>`
Unfortunately everytime I reload vscode, previously open views gets disconnected, i.e. any old ID cannot be reused again and the pdf must be opened in another view.
Is it possible to define a static ID=project1, so that  for the same view i can refer to
`http://<LOCAL_IP>:<PORT>/viewer.html?file=pdf..project1`

**Comment by James-Yu** [comment, marked as answer]
You can use a fixed port number with `latex-workshop.viewer.pdf.internal.port`. The encoded file path is not random.

**Reply by MarcoScaletta** [reply]
Thanks! I just went through the code now and saw the encoding.
